log/syslog.go

```diff
@@ -0,0 +1,86 @@
+// +build linux,!appengine darwin freebsd openbsd
+
+package log
+
+import (
+	"io"
+	"log/syslog"
+
+	"github.com/go-kit/kit/log/level"
+)
+
+type syslogWriter struct {
+	*syslog.Writer
+	selector func(keyvals ...interface{}) syslog.Priority
+}
+
+type SyslogAdapterOption interface {
+	Apply(*syslogWriter)
+}
+
+func NewSyslogWriter(w *syslog.Writer, options ...SyslogAdapterOption) io.Writer {
+	writer := &syslogWriter{
+		syslog.Writer: w,
+		selector:      defaultSyslogSelector,
+	}
+
+	for _, option := range options {
+		option.Apply(writer)
+	}
+
+	return *writer
+}
+
+type syslogWriterAdapter struct {
+	f func(string) error
+}
+
+func (a *syslogWriterAdapter) Write(b []byte) (int, error) {
+	return len(b), a.f(string(b))
+}
+
+func (w syslogWriter) GetSpecializedWriter(keyvals ...interface{}) io.Writer {
+	priority := w.selector(keyvals...)
+
+	switch priority {
```

Comment:
A complete implementation will of course handle all supported `syslog.Priority` values.